packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -174,11 +183,54 @@ export default class Sayt extends LitElement {
       return;
     }
 
-    const requestSaytResults = new CustomEvent('sfx::autocomplete_fetch_data', {
+    this.requestSaytAutocompleteTerms(query, searchbox);
+    this.requestSaytProducts(query, searchbox);
+  }
+
+  /**
+   * Dispatches an event to request data. Intended for requesting
+   * products or autocomplete terms.
+   *
+   * @param eventType The type of the event to be dispatched.
+   * @param query The query term.
+   * @param searchbox The ID of the associated searchbox.
+   */
+  dispatchRequestEvent(eventType: string, query: string, searchbox?: string) {
+    const requestEvent = new CustomEvent(eventType, {
       detail: { query, searchbox },
       bubbles: true
     });
-    window.dispatchEvent(requestSaytResults);
+    this.dispatchEvent(requestEvent);
+  }
+
+  /**
+   * Dispatches an [[AUTOCOMPLETE_REQUEST_RESULTS]] event with the provided data.
+   *
+   * @param query The search term to use.
+   * @param searchbox The optional searchbox ID associated with this search.
+   */
+  requestSaytAutocompleteTerms(query: string, searchbox?: string) {
+    this.dispatchRequestEvent(AUTOCOMPLETE_REQUEST_RESULTS, query, searchbox);
+  }
+
+  /**
+   * Dispatches an `sfx::sayt_products_request` event with the provided data.
```

Comment:
```suggestion
   * Dispatches an [[PRODUCTS_REQUEST_EVENT]] event with the provided data.
```

Reply:
I wasn't going to make any comments about event documentation because we can handle it in the events ticket, but sure ("an" should be changed to "a" if this change is made).

Reply:
I only mentioned it since the wrapper function for the autocomplete event handled it. Both were created in this ticket.